topdown/topdown_partial_test.go

```diff
@@ -223,7 +223,7 @@ func TestTopDownPartialEval(t *testing.T) {
 				`,
 			},
 			wantQueries: []string{
-				`y.bar = 1; z1 = input.foo[y]`,
+				`y.bar = 1; input.foo[y]`,
```

Comment:
(i) `z1 = input.foo[y]` and (ii) `input.foo[y]` are similar but not exactly the same. If `input.foo[y]` is the boolean value `false` then they'll have different answers.